doc/guides/maintaining-c-ares.md

```diff
@@ -0,0 +1,63 @@
+# Maintaining c-ares
+
+Updates to the c-ares dependency involve the following steps:
+1. Downloading the source archive for the new version.
+1. Unpacking the source in a temporary workspace directory.
+1. Removing the `test` directory (to save disk space).
+1. Copying over the existing `.gitignore`, pre-generated `config` directory and
+   `cares.gyp` files.
+1. Replacing the existing `deps/cares` with the workspace directory.
+1. Modifying the `cares.gyp` file for file additions/deletions.
+1. Rebuilding the main Node.js `LICENSE`.
+
+## Running the update script
+
+The `tools/update-cares.sh` script automates the update of the c-ares source
+files, preserving the existing files added by Node.js.
+
+In the following examples, `x.y.z` should match the c-ares version to update to.
+
+```console
+./tools/update-cares.sh x.y.z
+```
+
+e.g.
+```console
+./tools/update-cares.sh 1.18.1
+```
+
+## Check that Node.js still builds and tests
+
+It may be necessary to update `deps/cares/cares.gyp` if any significant changes
+have occurred upstream.
+
+## Rebuild the main Node.js license
+
+Run the `tools/license-builder.sh` script to rebuild the main Node.js `LICENSE`
+file. This may result in no changes if c-ares' license has not changed.
+
+```console
+./tools/license-builder.sh
+```
+
+If the `LICENSE` is updated for changes other than this c-ares update those
+should be done in a separate pull request first.
```

Comment:
I had to read this twice to understand the meaning. I believe "updated for changes other than this c-ares update" means changes to the `LICENSE` file that are caused by other dependencies? Maybe this can be clarified (but I don't have a simpler suggestion right now).

Reply:
I've reworded this to:
> If the updated `LICENSE` contains changes for other dependencies, those should
be done in a separate pull request first.